include/xtensor/xrepeat.hpp

```diff
@@ -0,0 +1,565 @@
+        using temporary_type = typename container_type::temporary_type;
+
+        static constexpr layout_type static_layout = xexpression_type::static_layout;
+        using bool_load_type = typename xexpression_type::bool_load_type;
+        using pointer = typename xexpression_type::pointer;
+        using const_pointer = typename xexpression_type::const_pointer;
+        using difference_type = typename xexpression_type::difference_type;
+
+        using iterable_type = xiterable<xrepeat<CT>>;
+        using stepper = typename iterable_type::stepper;
+        using const_stepper = typename iterable_type::stepper;
+
+        template<class CTA>
+        explicit xrepeat(CTA&& e, std::vector<std::ptrdiff_t>&& repeats, size_type axis);
+
+        template <class... Args>
+        reference operator()(Args... args);
+
+        template <class... Args>
+        const_reference operator()(Args... args) const;
+
+        const shape_type& shape() const noexcept;
+
+        template <class It>
+        reference element(It first, It last);
+
+        template <class It>
+        const_reference element(It first, It last) const;
+
+        stepper stepper_begin() const;
+
+        stepper stepper_begin(const shape_type& s) const;
+
+        stepper stepper_end(const layout_type l) const;
+
+        stepper stepper_end(const shape_type& s, const layout_type l) const;
+
+    private:
+        CT m_e;
+        const size_type m_repeating_axis;
+        const std::vector<std::ptrdiff_t> m_repeats;
+        shape_type m_shape;
+
+        reference access();
+
+        template <class Arg, class... Args>
+        reference access(Arg arg, Args... args);
+
+        const_reference access() const;
+
+        template <class Arg, class... Args>
+        const_reference access(Arg arg, Args... args) const;
+
+        template<size_type I, class Arg, class... Args>
+        inline const_reference access_impl(stepper&& s, Arg arg, Args... args) const
+        {
+            s.step(I, static_cast<size_type>(arg));
+            return access_impl<I+1>(std::forward<stepper>(s), args...);
+        }
+
+        template<size_type I>
+        inline const_reference access_impl(stepper&& s) const
+        {
+            return *s;
+        }
+
+        template<size_type I, class Arg, class... Args>
+        inline reference access_impl(stepper&& s, Arg arg, Args... args)
+        {
+            s.step(I, static_cast<size_type>(arg));
+            return access_impl<I+1>(std::forward<stepper>(s), args...);
+        }
+
+        template<size_type I>
+        inline reference access_impl(stepper&& s)
+        {
+            return *s;
+        }
+    };
+
+    template <class CT>
+    template <class CTA>
+    xrepeat<CT>::xrepeat(CTA&& e, std::vector<std::ptrdiff_t>&& repeats, size_type axis)
+        : m_e(std::forward<CTA>(e))
+        , m_repeating_axis(axis)
+        , m_repeats(std::forward<std::vector<std::ptrdiff_t>>(repeats))
```

Comment:
If you go with the overloaded API, this is fine, but the argument type should be a universal reference. Otherwise the argument type is fine, and `std::forward` should be replaced with `std::move`